running tests in parallel with `tox p` fails

Running tests in parallel with `tox p` fails, meaning I have to run all local test envs serially right now. It seems that the detected terminal environment is different in parallel mode, causing the pager tests to fail.

```
FAILED tests/test_utils.py::test_echo_via_pager[test0-cat] - AssertionError: Unexpected pager output in test case 'Plain string argument'
```

--- a/src/click/_compat.py
+++ b/src/click/_compat.py
@@ -12,6 +12,7 @@ from weakref import WeakKeyDictionary
 
 CYGWIN = sys.platform.startswith("cygwin")
 WIN = sys.platform.startswith("win")
+MAC = sys.platform == "darwin"
 auto_wrap_for_ansi: t.Callable[[t.TextIO], t.TextIO] | None = None
 _ansi_re = re.compile(r"\033\[[;?0-9]*[a-zA-Z]")
 

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -15,6 +15,7 @@ import pytest
 
 import click._termui_impl
 import click.utils
+from click._compat import MAC
 from click._compat import WIN
 from click._utils import UNSET
 
@@ -309,6 +310,10 @@ EchoViaPagerTest = namedtuple(
 
 
 @pytest.mark.skipif(WIN, reason="Different behavior on windows.")
+@pytest.mark.skipif(
+    MAC and sys.version_info >= (3, 13) and sys._is_gil_enabled(),
+    reason="Generator exception tests are flaky in Python 3.14t on macOS.",
+)
 @pytest.mark.parametrize(
     "pager_cmd", ["cat", "cat ", " cat ", "less", " less", " less "]
 )

